pkg/kubelet/images/image_manager.go

```diff
@@ -239,8 +213,18 @@ func (m *imageManager) EnsureImageExists(ctx context.Context, objRef *v1.ObjectR
 			}
 		}
 
-		pullRequired := m.imagePullManager.MustAttemptImagePull(requestedImage, imageRef, imagePullSecrets, imagePullServiceAccount)
-		if !pullRequired {
+		return imagePullSecrets, imagePullServiceAccount, nil
+	}
+
+	if imageRef != "" {
+		if !utilfeature.DefaultFeatureGate.Enabled(features.KubeletEnsureSecretPulledImages) {
```

Comment:
keeping this code block where it was before will simlpify the review. Took me some time to understand what happened here

Reply:
This is what the code originally looked like before https://github.com/kubernetes/kubernetes/pull/133079. I'd rather it stayed here.